src/gc-alloc-profiler.cpp

```diff
@@ -133,7 +135,9 @@ void _maybe_record_alloc_to_profile(jl_value_t *val, size_t size) JL_NOTSAFEPOIN
     profile.allocs.emplace_back(jl_raw_alloc_t{
         type,
         get_raw_backtrace(),
-        size
+        size,
+        jl_current_task,
+        cycleclock()
```

Comment:
Don't you need to use `jl_hrtime` aka `Base.time_ns`? Since the task pointer does not uniquely identify the task (as GC can recycle it), you'd need to correlate the allocation event with the window defined by the creation and finish times of the task to recover the accurate task identity. To join this data with the event data collected on the user-side Julia program (e.g., Dagger), it sounds more useful to use the clock available in Julia. Also, https://en.wikipedia.org/wiki/Time_Stamp_Counter suggests  `rdtsc` used for `cycleclock` on x86_64 does not seem to work as a global clock with different CPUs.

A bit aside, but I wonder if it's better to have a "true" task ID that is guaranteed be unique within a single process (e.g., prepare `const TASKIDS = [1:nthreads();]` and then issue `TASKIDS[threadid()] += nthreads()` in the task constructor in C). Although it's a bit annoying to have 64 more bits if you need it only for profiling...